lib/harmony/resources/resources.go

```diff
@@ -122,9 +122,19 @@ func getResources() (res Resources, err error) {
 		}
 	}
 
+	h, err := sysinfo.Host()
+	if err != nil {
+		return Resources{}, err
+	}
+
+	mem, err := h.Memory()
+	if err != nil {
+		return Resources{}, err
+	}
+
 	res = Resources{
 		Cpu: runtime.NumCPU(),
-		Ram: memory.FreeMemory(),
+		Ram: mem.Available,
```

Comment:
Yeah.. We could fix this by putting resource accounting in the DB, and using something like `/proc/sys/kernel/random/boot_id` to figure out what machine we're running on.

Obviously this is then made way more complicated with cgroups, which can make those resources seem independent, and one can then decide that it's fine to overprovision containers/cgroups..

Really I don't think there is a great way to prevent one from shooting themselves in the foot with resource accounting. I'd keep it as a very basic safety mechanism that is simple to understand and will prevent an unbounded number of let's say SDR processes running.

And really SDR is the only task for which the resource (memory) constraints are meaningful. All other tasks are either bounded by GPU count (TreeR/C/PoRep/PoSt snark) or they just don't use that much of any resource (moving around data).

This matters because SDR is also the only task type which you basically never run on more than one worker per machine, because all SDR implementations need to be able to assign themselves to very specific groups of threads, which only works within one process the way it's implemented today.

So this all means that: SDR is the only task where we'd really care about multiple providers per memory space AND sdr is the one task type which is basically never run as multiple providers per memory space, soo I don't think this is very big issue

Also I'd like to get SPs to configure their resource bounds with Max task settings; those allow for some degree of shooting yourself in the foot, but as long as the simple resource accounting provides some safety it shouldn't be too bad, and the 'max number of tasks' provides a much better mechanism for balancing pipelines imo (and SPs were asking for it since forever)